crates/iota-indexer/src/models/transactions.rs

```diff
@@ -54,12 +61,69 @@ pub struct TxGlobalOrder {
     pub optimistic_sequence_number: Option<i64>,
 }
 
-impl From<&IndexedTransaction> for TxGlobalOrder {
+impl From<&IndexedTransaction> for CheckpointTxGlobalOrder {
     fn from(tx: &IndexedTransaction) -> Self {
-        TxGlobalOrder {
+        Self {
             global_sequence_number: tx.tx_sequence_number as i64,
             tx_digest: tx.tx_digest.into_inner().to_vec(),
-            optimistic_sequence_number: Some(0),
+            index_status: Some(IndexStatus::Started),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Queryable, Insertable, QueryableByName, Selectable)]
+#[diesel(table_name = tx_global_order)]
+pub(crate) struct CheckpointTxGlobalOrder {
+    /// Number that represents the global ordering between optimistic and
+    /// checkpointed transactions.
+    ///
+    /// Optimistic transactions will share the same number as checkpointed
+    /// transactions. In this case, ties are resolved by the
+    /// `(global_sequence_number, optimistic_sequence_number)` pair that
+    /// guarantees deterministic ordering.
+    pub(crate) global_sequence_number: i64,
+    pub(crate) tx_digest: Vec<u8>,
+    /// The index status of checkpointed transactions.
+    ///
+    /// This essentially reserves the values `[0, -1]` of the
```

Comment:
I think we should mention in the docstring for TxGlobalOrder that -1 value also is allowed and what is it's meaning. Right now that docstring is a bit outdated.